services/src/datasets/storage.rs

```diff
@@ -0,0 +1,188 @@
+use crate::datasets::listing::DataSetProvider;
+use crate::error::Result;
+use crate::projects::project::LayerInfo;
+use crate::users::user::UserId;
+use crate::util::user_input::{UserInput, Validated};
+use async_trait::async_trait;
+use geoengine_datatypes::dataset::{DataSetId, DataSetProviderId, InternalDataSetId};
+use geoengine_datatypes::util::Identifier;
+use geoengine_operators::mock::MockDataSetDataSourceLoadingInfo;
+use geoengine_operators::source::OgrSourceDataset;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+use std::path::PathBuf;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct DataSet {
+    pub id: DataSetId,
+    pub name: String,
+    pub description: String,
+    pub data_type: LayerInfo, // TODO: custom type instead of reusing existing one?
+    pub source_operator: String, // TODO: enum?
+}
+
+impl From<AddDataSet> for DataSet {
+    fn from(value: AddDataSet) -> Self {
+        Self {
+            id: DataSetId::Internal(InternalDataSetId::new()),
```

Comment:
maybe add a TODO for using staging later on

Reply:
the idea is that added data sets (`AddDataSet`) always turn into internal data sets. For staging you would first `stage` and then `import` but this will be a different PR